account_loan/models/account_loan.py

```diff
@@ -370,15 +370,27 @@ def _compute_posted_lines(self):
         """
         Recompute the amounts of not finished lines. Useful if rate is changed
         """
-        amount = self.loan_amount
-        for line in self.line_ids.sorted("sequence"):
-            if line.move_ids:
-                amount = line.final_pending_principal_amount
-            else:
-                line.rate = self.rate_period
-                line.pending_principal_amount = amount
-                line._check_amount()
-                amount -= line.payment_amount - line.interests_amount
+        self.ensure_one()
+        posted = self.line_ids.filtered(lambda line: line.move_ids)
+        unposted = self.line_ids.filtered(lambda line: not line.move_ids)
+        self.fixed_loan_amount = self.pending_principal_amount
+        self.fixed_periods = self.periods - len(posted)
+        seq = max(posted.mapped("sequence"), default=0) + 1
+        date = max(
+            posted.mapped("date"), default=self.start_date or datetime.today().date()
+        )
+        amount = self.fixed_loan_amount
+        initial_date = date
+        delta = relativedelta(months=self.method_period)
+        if not self.payment_on_first_period:
+            date = initial_date + delta
+            initial_date = date
+        for i, line in enumerate(unposted[: self.fixed_periods]):
```

Comment:
Instead of doing this, why not doing one general for and if the condition (I suppose `line.move_ids"), do `continue` inside the for?

Reply:
The number of lines and the multiple flows says the contrary, as you have several intermediate variables and weird slices that are not very readable.